tests/test_launcher.py

```diff
@@ -192,3 +194,47 @@ def test_server_ip(server_type_remote_process):
     assert server_type_remote_process.info["server_ip"] is not None
     assert server_type_remote_process.info["server_port"] is not None
     assert server_type_remote_process.info["server_version"] is not None
+
+
+@pytest.mark.skipif(
+    DPF_SERVER_TYPE is not None,
+    reason="This test is for a run with default server type",
+)
+def test_start_with_dpf_server_type_env():
+    dpf_server_type_str = "DPF_SERVER_TYPE"
+    try_serv_type = os.environ.get(dpf_server_type_str, None)
+    if try_serv_type:
+        raise Exception(
+            "Fixture is not correctly working"
+        )  # a specific case is already set to run the unit tests
+    else:
+        if SERVERS_VERSION_GREATER_THAN_OR_EQUAL_TO_4_0:
+            # test for v222 and higher
+            os.environ[dpf_server_type_str] = "GRPC"
+            my_serv = core.start_local_server(as_global=False)
```

Comment:
@anslpa Are we sure all servers will correctly close?

Reply:
I added "shutdown()" calls for the started servers.